example/src/App.tsx

```diff
@@ -194,16 +181,16 @@ const ResultItem = React.forwardRef(
     ref: React.Ref<HTMLDivElement>
   ) => {
     const ancestors = React.useMemo(() => {
-      return (function collect(action: ActionImpl, ancestors = []) {
-        if (action.parent && action.parent.id !== currentRootActionId) {
-          ancestors.push(action.parent);
-          if (action.parent.parent) {
-            collect(action.parent.parent, ancestors);
-          }
-        }
-        return ancestors;
-      })(action);
-    }, [action, currentRootActionId]);
+      if (!currentRootActionId) return action.ancestors;
+      const index = action.ancestors.findIndex(
+        (ancestor) => ancestor.id === currentRootActionId
+      );
+      // +1 removes the currentRootAction; e.g.
+      // if we are on the "Set theme" parent action,
+      // the UI should not display "Set theme… > Dark"
+      // but rather just "Dark"
+      return action.ancestors.slice(index + 1);
+    }, [action.ancestors, currentRootActionId]);
```

Comment:
It would be wonderful to have this logic handled within `ActionImpl`, but `ActionImpl` shouldn't have context of `currentRootActionId`.